examples/websockets/src/main.rs

```diff
@@ -59,58 +76,169 @@ async fn main() {
     let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
     tracing::debug!("listening on {}", addr);
     axum::Server::bind(&addr)
-        .serve(app.into_make_service())
+        .serve(app.into_make_service_with_connect_info::<SocketAddr>())
         .await
         .unwrap();
 }
 
+/// A formally required handler for the request.
```

Comment:
What does "formally" mean here?

Reply:
Well every websocket open is an HTTP request, so before we can get to websocket stuff proper we need to have a handler for the original HTTP request, which that function is. Which is why it is the last place where HTTP headers can be seen, and after that we are essentially talking in another protocol. In this sense it is a "formality" if all you wanted was websockets. I can try to rephrase this.

Reply:
Ah I see, makes sense. Maybe something like "handle upgrading the connection to use websockets" or something?

Reply:
"switching from HTTP to websocket protocol" wouldn't be bad either.

Reply:
Maybe a bit verbose, but I think it is correct now:
```
/// The handler for the HTTP request (this gets called when the HTTP GET lands at the start
/// of websocket negotiation. After this completes, the actual switching from HTTP to
/// websocket protocol will occur.
/// This is the last point where we can extract TCP/IP metadata such as IP address of the client
/// as well as things from HTTP headers such as user-agent of the browser etc.
```